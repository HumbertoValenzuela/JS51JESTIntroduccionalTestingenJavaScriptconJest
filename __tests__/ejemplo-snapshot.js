// 6. Introducción a Snapshots

const cliente = {
    // Si tienes el snapshot de este objeto. Luego lo modificas por 
    // ej: nombre: 'Humberto 2' esto dará un error

    // ¿Cómo solucionar este problema? Se actualiza el snapshot con los
    // nuevos datos. En la Terminal escribir npm t -- -u
    // Otras formas no recomendadas: eliminar el archivo o modificar
    nombre: 'Humberto Valenzuela Actualizado',
    balance: 900000,
    tipo: 'Premium'
}

describe('Testing al Cliente', () => {
    test('Es Humberto', () => {
        expect(cliente).toMatchSnapshot();
        // Esto creará una carpeta __snapshot__
        // Los archivos dentro de __snapshot__ son manejados por JEST
    });
});