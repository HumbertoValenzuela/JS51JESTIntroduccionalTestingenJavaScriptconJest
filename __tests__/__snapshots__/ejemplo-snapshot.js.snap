// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing al Cliente Es Humberto 1`] = `
Object {
  "balance": 900000,
  "nombre": "Humberto Valenzuela Actualizado",
  "tipo": "Premium",
}
`;
